### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by kendallstrautman**
Updates most dependencies and most importantly, to Next 13. If we choose to refactor to app router, that will be a separate PR. Also, I'd prefer to separate out refactors or stylistic code changes in a subsequent PR to reduce the scope of this upgrade.

Since I changed the prettier config to use tabs instead of spaces, I suggest viewing the diff [without whitespace changes](https://github.com/hashicorp/mktg-webdev-challenge/pull/47/files?diff=split&w=1).

### CODE REVIEW

File: pages/home/index.tsx [hunk 1]
```diff
@@ -1,29 +1,40 @@
 import fs from 'fs'
 import path from 'path'
+import highlight from '@mapbox/rehype-prism'
+import rehypeSurfaceCodeNewlines from '@hashicorp/platform-code-highlighting/rehype-surface-code-newlines'
 import { serialize } from 'next-mdx-remote/serialize'
 import { MDXRemote, MDXRemoteSerializeResult } from 'next-mdx-remote'
 import { GetStaticPropsResult } from 'next'
-import markdownDefaults from '@hashicorp/platform-markdown-utils'
```
Comment (kendallstrautman) [inline pages/home/index.tsx:6]: Our markdown defaults aren't fully compatible with mdx V2. So I imported the plugins we need directly.

File: .eslintrc.js [hunk 2]
```diff
@@ -1,6 +1,25 @@
-const eslintConfig = require('@hashicorp/platform-cli/config/.eslintrc.js')
 module.exports = {
```
Comment (kendallstrautman) [inline .eslintrc.js:1]: Copied our current settings from dev portal to align with our standard lint rules.

File: layouts/base/components/footer/components/newsletter-signup-form/index.js [hunk 3]
```diff
@@ -1,6 +1,4 @@
-import React, { useState, useEffect } from 'react'
-
-import getAnalytics from './helpers/getAnalytics'
```
Comment (kendallstrautman) [inline layouts/base/components/footer/components/newsletter-signup-form/index.js:3]: This wasn't called, not needed.

File: layouts/base/components/footer/index.js [hunk 4]
```diff
@@ -13,391 +13,399 @@ import IconYouTube from './social/icon-youtube'
...
+								title="Meetup"
+								data-ga-footer="Social | Meetup"
+							>
+								<IconMeetup />
+							</a>
+						</li>
+					</ul>
+				</div>
+				<div className="primary-section">
+					<ul className="solutions">
```
Comment (kendallstrautman) [inline layouts/base/components/footer/index.js:102]: All updates from legacy link behavior in this file.

File: layouts/base/components/nav/index.jsx [hunk 5]
```diff
@@ -23,662 +23,647 @@ import Image from '@hashicorp/react-image'
...
-          </a>
-        </Link>
+	return (
+		<header className="g-nav g-type-body-strong">
+			{activeDesktop || activeMobile ? (
+				<div className="background" onClick={closePanels} />
+			) : undefined}
+			<nav className={`${activeDesktop || activeMobile ? 'active' : ''}`}>
+				{/* Logos */}
+				<Link
```
Comment (kendallstrautman) [inline layouts/base/components/nav/index.jsx:140]: More legacy link behavior updates here.

File: pages/_app.tsx [hunk 6]
```diff
@@ -1,26 +1,20 @@
 import React from 'react'
 import { AppProps } from 'next/app'
 import './style.css'
-import '@hashicorp/platform-util/nprogress/style.css'
-
-import NProgress from '@hashicorp/platform-util/nprogress'
-import Router from 'next/router'
-
-NProgress({ Router })
```
Comment (kendallstrautman) [inline pages/_app.tsx:9]: This isn't necessary and wasn't even rendering.

**Review by BrandonRomano — APPROVED** [review]
Looks good!  I am getting some warning RE: peer dependencies when running `npm i`, but it is not a big problem.

**Pull request merged**